codegen/src/main/java/io/spine/tools/java/FileName.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.java;
+
+import com.google.protobuf.DescriptorProtos.DescriptorProto;
+import com.google.protobuf.DescriptorProtos.EnumDescriptorProto;
+import com.google.protobuf.DescriptorProtos.ServiceDescriptorProto;
+import io.spine.type.StringTypeValue;
+
+import static io.spine.tools.CodePreconditions.checkNotEmptyOrBlank;
+
+/**
+ * A Java file name.
+ *
+ * @author Alexander Yevsyukov
+ */
+public final class FileName extends StringTypeValue {
+
+    private static final String GRPC_CLASSNAME_SUFFIX = "Grpc";
+    private static final String EXTENSION = ".java";
+
+    private FileName(String value) {
+        super(value);
+    }
+
+    /**
+     * Obtains the name for the passed type.
+     */
+    public static FileName forType(String typeName) {
```

Comment:
Since it is a **Java** file name, can we use `Class` instead of `String`?

If so, the implementation may become not so simplistic (e.g. support for inner types etc.).

Reply:
If the class is not yet generated when referring to its name, this comment should not be applied.

Reply:
Since we're dealing mostly with code generation, we don't have a class yet.